dll/kernel32.cpp

```diff
@@ -1801,42 +1801,45 @@ namespace kernel32 {
 	}
 
 	std::string str_for_LCType(int LCType) {
-		std::string ret;
 		// https://www.pinvoke.net/default.aspx/Enums/LCType.html
 		if (LCType == 4100) { // LOCALE_IDEFAULTANSICODEPAGE
 			// Latin1; ref GetACP
-			ret = "28591";
+			return "28591";
 		}
 		if (LCType == 4097) { // LOCALE_SENGLANGUAGE
-			ret = "Lang";
+			return "Lang";
 		}
 		if (LCType == 4098) { // LOCALE_SENGCOUNTRY
-			ret = "Country";
+			return "Country";
 		}
-		return ret;
+		assert(false); // something better?
 	}
 
-	int WIN_FUNC GetLocaleInfoA(unsigned int Locale, int LCType, char *lpLCData, int cchData) {
+	int WIN_FUNC GetLocaleInfoA(unsigned int Locale, int LCType, LPSTR lpLCData, int cchData) {
 		DEBUG_LOG("GetLocaleInfoA %d %d\n", Locale, LCType);
 		std::string ret = str_for_LCType(LCType);
+		size_t len = ret.size() + 1;
 
 		if (!cchData) {
-			return ret.size() + 1;
+			return len;
 		} else {
-			memcpy(lpLCData, ret.c_str(), ret.size() + 1);
+			assert(len <= cchData);
+			memcpy(lpLCData, ret.c_str(), len);
 			return 1;
 		}
 	}
 
-	int WIN_FUNC GetLocaleInfoW(unsigned int Locale, int LCType, wchar_t *lpLCData, int cchData) {
+	int WIN_FUNC GetLocaleInfoW(unsigned int Locale, int LCType, LPWSTR lpLCData, int cchData) {
 		DEBUG_LOG("GetLocaleInfoW %d %d\n", Locale, LCType);
 		std::string info = str_for_LCType(LCType);
-		std::wstring ret(ret.begin(), ret.end());
+		LPWSTR ret = stringToWideString(info.c_str());
```

Comment:
For some reason we made this return a raw pointer, which needs to be deleted afterwards to not get memory leaks. You can either make it return vector<uint16_t> instead to not have that problem (would be nice), or be lazy and do the delete

(oh, and also: good thing to get rid of wstring, I forgot to bring up another thing about it which is that it's 32 bits on Linux, so the code didn't work at all)

Reply:
yeah let me just do the lazy thing and forever be known as the lazy pointer deleter /s